### DISCUSSION TITLE
在bash脚本中对字符串末尾追加字符串是否有快速的方法？（不是O(n)级别的

### DISCUSSION THREAD

**Discussion by DM0U** [physics-data/faq#107] [Q&A] [answered]
在各种编程语音如C++中，`str += str_add`的运行速度将明显快于`str = str + str_add`。请问bash中是否有类似的实现？

具体来说，在bash中，如果我明确只想将`str_add`追加至`str`尾部，是否存在方法，其在程序底层逻辑上直接在`str`之后的空间直接写入`str_add`，而不是先计算`“${str}${str_add}”`，之后再赋值给str变量?
<br/>
<br/>
以及如果不行的话，用`sed`命令和用`“${str}${str_add}”`哪个更快？

**Comment by Harry-Chen** [comment, marked as answer]
很遗憾，不存在这样的写法。事实上，其它语言中，也不能做到你说的 “在程序底层逻辑上直接在str之后的空间直接写入str_add”，事实上大多数语言中这样做会引起内存的重新分配。类似地，后一种写法也未必更低效，大多数编译器能够有效地消除其中多余的字符串拷贝。

对于你的问题，在 bash 中直接拼接变量将明显快于使用外部程序 `sed`。
